calcom/cal.com
# fix: v2 event-types & MS outlook event description

Releasing this 2 fixes in same PR because needed to update platform libraries for both and otherwise would have to wait for fix 1 merge.

# Problem 1 - event-types
If an event-type was created using version 2024_04_15  (prior to refactor #15457  / version 2024_06_14), then `bookingFields` were populated with system fields.

The new version 2024_06_14 only adds booking fields passed by the user to give full control. However, if user then uses version 2024_06_14 to fetch event-types created using version 2024_04_15, then error ocurred.

## Reason
2024_04_15 even-type booking fields includes user and system event-types, but 2024_06_14 used zod schema to parse and return only user event-types, so if system was encountered the zod schema failed and error occurred:
```
{
    "status": "error",
    "timestamp": "2024-08-04T19:53:41.713Z",
    "path": "/v2/organizations/15359/teams/17166/event-types",
    "error": {
        "code": "BAD_REQUEST",
        "message": "6.editable - Invalid literal value, expected \"user\", 6.sources.0.fieldRequired - Invalid literal value, expected true, 6.sources.0.label - Invalid literal value, expected \"User\", 6.sources.0.type - Invalid literal value, expected \"user\", 6.sources.0.id - Invalid literal value, expected \"user\", 6.label - Required, 5.editable - Invalid literal value, expected \"user\", 5.sources.0.fieldRequired - Invalid literal value, expected true, 5.sources.0.label - Invalid literal value, expected \"User\"
    }
}
```

Why did we change this logic? Because if user does not provide any booking fields when creating an event-type, then in database there should not be bunch of default system fields that would be also returned when fetching the event-type. Furthermore, booking atom then transforms api response by adding name and e-mail booking fields by default, so `bookingFields` acts as custom booking fields user needs.

## Solution

Shortly - make event-types with booking fields containing system booking fields backwards compatible with the new event-types 2024_06_14 version.

1. In commit [refactor: have system and user booking fields schemas and united schema](https://github.com/calcom/cal.com/commit/0e6fd0e3b8789dfa8d888b5337092b416bba4218) simplify code to clearly distinguish between system and user booking fields.
2. In commit [fix: event-types output service filter out system booking fields](https://github.com/calcom/cal.com/commit/92676ea18e79c0eade6b28769a68b6665756dd25) when constructing event-type response filter out system fields.

Chore: update platform libraries to support it and publish to version `0.0.25`

## Test
This issue was reported by our customer in [Slack](https://calcominc.slack.com/archives/C06QJ0P3EKH/p1722801313901169) and we checked the faulty event-type that has the booking fields below:
```
[{"name": "name", "type": "name", "sources": [{"id": "default", "type": "default", "label": "Default"}], "editable": "system", "required": true, "defaultLabel": "your_name"}, {"name": "email", "type": "email", "sources": [{"id": "default", "type": "default", "label": "Default"}], "editable": "system", "required": true, "defaultLabel": "email_address"}, {"name": "location", "type": "radioInput", "label": "", "sources": [{"id": "default", "type": "default", "label": "Default"}], "editable": "system", "required": false, "placeholder": "", "defaultLabel": "location", "getOptionsAt": "locations", "optionsInputs": {"phone": {"type": "phone", "required": true, "placeholder": ""}, "attendeeInPerson": {"type": "address", "required": true, "placeholder": ""}}, "hideWhenJustOneOption": true}, {"name": "title", "type": "text", "hidden": true, "sources": [{"id": "default", "type": "default", "label": "Default"}], "editable": "system-but-optional", "required": true, "defaultLabel": "what_is_this_meeting_about", "defaultPlaceholder": ""}, {"name": "notes", "type": "textarea", "sources": [{"id": "default", "type": "default", "label": "Default"}], "editable": "system-but-optional", "required": false, "defaultLabel": "additional_notes", "defaultPlaceholder": "share_additional_notes"}, {"name": "guests", "type": "multiemail", "hidden": false, "sources": [{"id": "default", "type": "default", "label": "Default"}], "editable": "system-but-optional", "required": false, "defaultLabel": "additional_guests", "defaultPlaceholder": "email"}, {"name": "rescheduleReason", "type": "textarea", "views": [{"id": "reschedule", "label": "Reschedule View"}], "sources": [{"id": "default", "type": "default", "label": "Default"}], "editable": "system-but-optional", "required": false, "defaultLabel": "reason_for_reschedule", "defaultPlaceholder": "reschedule_placeholder"}]
```

1. Create platform team event-type and pass these as `bookingFields`
2. Make a request to `api/v2/organizations/:orgId/teams/:teamId/event-types` and see that `bookingFields` are returned as `[]` aka with system fields filtered out.

# Problem 2 - outlook event description formatting

Events scheduled in outlook do not have newlines:
<img width="1177" alt="Screenshot 2024-08-08 at 15 39 58" src="https://github.com/user-attachments/assets/94986a30-b2a0-4aba-82d2-2131d594e6b4">

Solution: In [office365calendar/lib/CalendarService.ts](https://github.com/calcom/cal.com/pull/16132/files#diff-4f075841987a4d1474327cb5c6097757822d5df69616798d8014ff3207a3034b) replace newline characters with `<br>` since the content is sent as HTML. The description now is displayed just like in google calendar events:
<img width="1158" alt="Screenshot 2024-08-08 at 15 51 34" src="https://github.com/user-attachments/assets/d2855e71-0635-41ed-b72e-413b82e48010">
